.github/workflows/build.yml

```diff
@@ -21,6 +21,25 @@ jobs:
       - name: Run Biome
         run: biome ci .
 
+  check-yarn-lock:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Check for yarn.lock changes
+        run: |
+          git diff --exit-code yarn.lock || (echo "yarn.lock has changed. Commit the updated file." && exit 1)
```

Comment:
This is probably OK but there's also https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-frozen-lockfile, which is the built-in way of doing it. Could be worth using that